### PR TITLE
Replace `windows-sys` by `extern` call

### PR DESCRIPTION
**Pull request by yozhgoor**

### CODE REVIEW

File: src/subprocess.rs [hunk 1]
```diff
@@ -215,42 +212,54 @@ mod windows {
...
+                    let wide_path: Vec<u16> =
+                        path.as_os_str().encode_wide().chain(once(0)).collect();
                     wide_path.as_ptr()
                 })
-                .unwrap_or(std::ptr::null_mut());
+                .unwrap_or(null_mut());
 
             let command = command.encode_wide().collect::<Vec<_>>();
 
+            extern "system" {
```
Comment (cecton) [inline src/subprocess.rs:230]: dont we usually place them at tthe beginning of the file or in its own file?

File: src/subprocess.rs [hunk 2]
```diff
@@ -303,12 +320,12 @@ mod windows {
                     &mut exit_code as *mut u32,
                 ) != 0
                 {
-                    if exit_code as i32 == STATUS_PENDING {
+                    if exit_code as i32 == 0x00000103 {
```
Comment (cecton) [inline src/subprocess.rs:323]: make a const?

**Review by cecton — CHANGES_REQUESTED** [review]

File: src/binding.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+pub const INFINITE: u32 = 0xFFFFFFFF;
+pub const WAIT_OBJECT_0: u32 = 0x00000000;
+pub const STATUS_PENDING: u32 = 0x00000103;
+
+pub type PCWSTR = *const u16;
+pub type BOOL = i32;
+pub type DWORD = u32;
+pub type PDWORD = *mut u32;
+pub type PWSTR = *mut u16;
+pub type UINT = u32;
```
Comment (cecton) [inline src/binding.rs:14]: You probably don't want anything from binding.rs to be public

File: src/project.rs [hunk 4]
```diff
@@ -95,8 +95,12 @@ impl Project {
 
             #[cfg(windows)]
             if config.editor.is_some() {
+                extern "system" {
+                    fn FreeConsole() -> i32;
+                }
```
Comment (cecton) [inline src/project.rs:100]: You forgot to move that one to binding.rs

**Review by cecton — CHANGES_REQUESTED** [review]

File: src/binding.rs (hunk 3 above)
Comment (yozhgoor) [reply src/binding.rs]: You meant replace it by `pub(crate)` right?

Comment (cecton) [reply src/binding.rs]: Yeah pub(crate) is kinda private to the crate. I don't why it's called pub xD

**Review by cecton — APPROVED** [review]
Excellent work! :ok_hand: and nicely documented

**Pull request merged**